readthedocs/core/models.py

```diff
@@ -43,6 +43,15 @@ class UserProfile(TimeStampedModel):
     whitelisted = models.BooleanField(_("Whitelisted"), default=False)
     banned = models.BooleanField(_("Banned"), default=False)
 
+    # Opt-out on emails
+    # NOTE: this is a temporary field that we can remove after September 25, 2023
+    # See https://blog.readthedocs.com/migrate-configuration-v2/
+    optout_email_config_file_deprecation = models.BooleanField(
+        _("Opt-out from email about 'Config file deprecation'"),
+        default=False,
+        null=True,
+    )
+
```

Comment:
I know that the approach is to keep this minimal, but if we stored this in a model, we could expand it in future deprecations?

This could also make it easier to remove data afterwards. We just delete stuff from a table, rather than having migrations to remove columns.

```
class NotificationPreferences(models.Model):
    user_profile = models.ForeignKey(...)
    email_type = models.Charfield=(choices=[("config_file_required", "Deprecation of config file")])
```

Reply:
The addition to this is that we don't need a form, we just need a view that automatically saves the preference and returns a "you will no longer receive this notification" response.

Reply:
Yeah, I just didn't want to invest too much time on thinking and modeling the future right now. I'm not sure what are going to be our requirements yet and I prefer to stick with the simplest approach.

I'd like to get some experience with this approach before building the generic/expandable one. The more we think about this, the more time it will take to be deployed, and the less useful it will be for this particular case because we will be still sending emails to people that don't want those emails.

I think it's a good idea to consider in the near future, tho, where more deprecation are coming.